warehouse/clickhouse/clickhouse.go

```diff
@@ -562,6 +585,75 @@ func (ch *HandleT) loadTable(tableName string, tableSchemaInUpload warehouseutil
 	return
 }
 
+func (ch *HandleT) credentials() (accessKeyID, secretAccessKey string, err error) {
+	if ch.ObjectStorage == warehouseutils.S3 {
+		return warehouseutils.GetConfigValue(warehouseutils.AWSAccessSecret, ch.Warehouse), warehouseutils.GetConfigValue(warehouseutils.AWSAccessKey, ch.Warehouse), nil
+	}
+	if ch.ObjectStorage == warehouseutils.MINIO {
+		return warehouseutils.GetConfigValue(warehouseutils.MinioAccessKeyID, ch.Warehouse), warehouseutils.GetConfigValue(warehouseutils.MinioSecretAccessKey, ch.Warehouse), nil
+	}
+	return "", "", errors.New("objectStorage not supported for loading using S3 engine")
+}
+
+func (ch *HandleT) loadByCopyCommand(tableName string, tableSchemaInUpload warehouseutils.TableSchemaT) (err error) {
+	pkgLogger.Infof("LoadTable By COPY command Started for table: %s", tableName)
+
+	strKeys := warehouseutils.GetColumnsFromTableSchema(tableSchemaInUpload)
+	sort.Strings(strKeys)
+	sortedColumnNames := strings.Join(strKeys, ",")
+	sortedColumnNamesWithDataTypes := warehouseutils.JoinWithFormatting(strKeys, func(idx int, name string) string {
+		return fmt.Sprintf(`%s %s`, name, rudderDataTypesMapToClickHouse[tableSchemaInUpload[name]])
+	}, ",")
+
+	csvObjectLocation, err := ch.Uploader.GetSampleLoadFileLocation(tableName)
+	if err != nil {
+		err = fmt.Errorf("failed to get sample load file location with error: %s", err.Error())
+		return
+	}
+	loadFolder := csvObjectLocation
+	loadFolder = loadFolder[:strings.LastIndex(loadFolder, "/")]
+	loadFolder = loadFolder + "/*.csv.gz"
+
+	accessKeyID, secretAccessKey, err := ch.credentials()
+	if err != nil {
+		err = fmt.Errorf("failed to get credentials during load for copy with error: %s", err.Error())
+		return
+	}
+
+	sqlStatement := fmt.Sprintf(`
+		INSERT INTO %[1]q.%[2]q (
+			%[3]s
+		)
+		SELECT
+		  *
+		FROM
+		  s3(
+			'%[4]s',
+		  	'%[5]s',
+		  	'%[6]s',
+			'CSV',
+			'%[7]s',
+			'gz'
+		  ) settings date_time_input_format = 'best_effort';
+		`,
+		ch.Namespace,                   // 1
+		tableName,                      // 2
+		sortedColumnNames,              // 3
+		loadFolder,                     // 4
+		accessKeyID,                    // 5
+		secretAccessKey,                // 6
+		sortedColumnNamesWithDataTypes, // 7
+	)
+	_, err = ch.Db.Exec(sqlStatement)
```

Comment:
Regex replaces here. We should not logs the SQL statement.

Reply:
Do you think we could use prepared statements here? For:
```
		loadFolder,                     // 4
		accessKeyID,                    // 5
		secretAccessKey,                // 6
		sortedColumnNamesWithDataTypes, // 7
		```